app/models/user.rb

```diff
@@ -3,5 +3,13 @@ class User < ActiveRecord::Base
 
   has_many :links
 
-  validates_uniqueness_of :email
+  validates :name,  presence: true, length: { maximum: 50 }
+
+  VALID_EMAIL_REGEX = /\A[\w+\-.]+@[a-z\d\-.]+\.[a-z]+\z/i
```

Comment:
Cool..

On Thu, Apr 9, 2015 at 1:15 PM, Ben Woodall <notifications@github.com>
wrote:

> In app/models/user.rb
> <https://github.com/benwoody/shrtnr/pull/4#discussion_r28098226>:
>
> > @@ -3,5 +3,13 @@ class User < ActiveRecord::Base
> >
> >    has_many :links
> >
> > -  validates_uniqueness_of :email
> > +  validates :name,  presence: true, length: { maximum: 50 }
> > +
> > +  VALID_EMAIL_REGEX = /\A[\w+\-.]+@[a-z\d\-.]+\.[a-z]+\z/i
>
> I might talk about this in class, but validating emails via regex is a
> no-no. Sort of an unwritten rule.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/benwoody/shrtnr/pull/4/files#r28098226>.
>



-- 
Joe Simpson
“Reasonable people adapt themselves to the world. Unreasonable people
attempt to adapt the world to themselves. All progress, therefore, depends
on unreasonable people.”
George Bernard Shaw